pkg/goanalysis/runner_loadingpackage.go

```diff
@@ -50,24 +50,30 @@ func (lp *loadingPackage) analyzeRecursive(stopChan chan struct{}, loadMode Load
 
 		for _, imp := range lp.imports {
 			go func(imp *loadingPackage) {
-				imp.analyzeRecursive(stopChan, loadMode, loadSem)
+				imp.analyzeRecursive(ctx, cancel, loadMode, loadSem)
 
 				wg.Done()
 			}(imp)
 		}
 
 		wg.Wait()
 
-		lp.analyze(stopChan, loadMode, loadSem)
+		lp.analyze(ctx, cancel, loadMode, loadSem)
 	})
 }
 
-func (lp *loadingPackage) analyze(stopChan chan struct{}, loadMode LoadMode, loadSem chan struct{}) {
+func (lp *loadingPackage) analyze(ctx context.Context, cancel context.CancelFunc, loadMode LoadMode, loadSem chan struct{}) {
 	loadSem <- struct{}{}
 	defer func() {
 		<-loadSem
 	}()
 
+	select {
+	case <-ctx.Done():
+		return
+	default:
+	}
```

Comment:
It should be logically the same but when `ctx.Done()` is closed `ctx.Err()` will return a non nil error. Just a personal pref on what's easier to read I guess.


```suggestion
	if ctx.Err() != nil {
		return
	}
```